hack/generated/controllers/suite_test.go

```diff
@@ -124,22 +124,45 @@ func teardown() error {
 	return nil
 }
 
-func TestMain(m *testing.M) {
+type controllerTest struct {
+	name string
+	test func(ctx ControllerTestContext, t *testing.T)
+}
 
-	err := setup()
-	if err != nil {
-		panic(err)
+func runTests(t *testing.T, tests []controllerTest) {
+	ctx, setupErr := setup()
+	if setupErr != nil {
+		panic(setupErr)
 	}
 
-	// call flag.Parse() here if TestMain uses flags
-	code := m.Run()
+	defer func() {
+		if teardownErr := teardown(ctx); teardownErr != nil {
+			panic(teardownErr)
+		}
+	}()
 
-	err = teardown()
-	if err != nil {
-		panic(err)
+	for _, test := range tests {
+		if !t.Run(test.name, func(t *testing.T) { test.test(ctx, t) }) {
+			return
+		}
+	}
+}
+
+func Test_Controller_Integrations_Slow(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping slow tests in short mode")
 	}
 
-	os.Exit(code)
+	tests := []controllerTest{
+		{"ResourceGroup CRUD", Integration_ResourceGroup_CRUD},
```

Comment:
kinda annoying that we have to add stuff here when we add a test. I wonder if we could discover them via reflection?

Reply:
no, there is no way to reflect over packages in Go